lib/seattleflu/db/cli/command/etl/manifest.py

```diff
@@ -88,48 +90,170 @@ def etl_manifest(*, action: str):
+                        mark_skipped(db, manifest_record.id)
+                        continue
+
+                    assert not collection_identifier \
+                        or collection_identifier.set_name in expected_identifier_sets["collections"], \
+                            f"Collection identifier found in set «{collection_identifier.set_name}», not {expected_identifier_sets['collections']}"
+
+                    # Upsert sample cooperatively with enrollments ETL routine
+                    #
+                    # The details document was intentionally modified by two pop()s
+                    # earlier to remove barcodes that were looked up.  The
+                    # rationale is that we want just one clear place in the
+                    # warehouse for each piece of information.
+                    sample, status = upsert_sample(db,
+                        identifier            = sample_identifier.uuid,
+                        collection_identifier = collection_identifier.uuid if collection_identifier else None,
+                        additional_details    = manifest_record.document)
+                    
+                    mark_loaded(db, manifest_record.id,
                     status = status,
                     sample_id = sample.id)
+                
+                else:
+                    # Convert kit barcode to full identifier and check 
+                    # it's from the correct identifier set.
+                    kit_barcode = manifest_record.document.pop("sample")
+                    kit_identifier = check_barcode(db, 
+                        record          = manifest_record, 
+                        barcode         = kit_barcode, 
+                        identifier_set  = expected_identifier_sets["kits"])
+
+                    # The only time a kit_identifier is not returned is if
+                    # there was no identifier that matches the barcode.
+                    # In this case, the record "skipped".
+                    if not kit_identifier: continue
+
+                    # Convert UTM collection barcode to full identifier and 
+                    # check it's from the correct identifier set.
+                    utm_collection_barcode = manifest_record.document.pop("utm", None)
+                    utm_sample_id = None
+                    utm_status = "None"
+                    
+                    if utm_collection_barcode:
+
+                        utm_collection_identifier = check_barcode(db, 
+                            record          = manifest_record,
+                            barcode         = utm_collection_barcode,
+                            identifier_set  = expected_identifier_sets["kit_collections"])
+
+                        # Skip record if no matching UTM collection identifier
+                        # found for the UTM collection barcode    
+                        if not utm_collection_identifier: continue
+                        
```

Comment:
Are these (and line 173 below) LOG-worthy?

Reply:
These are logged within `check_barcode` on line 378. Although the log message is not very specific to which barcode is unknown...

Reply:
How many barcodes do these things have? :anguished:

Reply:
Kits can have up to 5 barcodes (kit, utm collection, utm sample, rdt collection, test strip)